.ci/magician/cmd/request_service_reviewers.go

```diff
@@ -0,0 +1,149 @@
+/*
+* Copyright 2023 Google LLC. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ */
+package cmd
+
+import (
+	"fmt"
+	"magician/github"
+	"math/rand"
+	"os"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/magic-modules/tools/issue-labeler/labeler"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+)
+
+// requestServiceReviewersCmd represents the requestServiceReviewers command
+var requestServiceReviewersCmd = &cobra.Command{
+	Use:   "request-service-reviewers PR_NUMBER",
+	Short: "Assigns reviewers based on the PR's service labels.",
+	Long: `This command requests (or re-requests) review based on the PR's service labels.
+
+	If a PR has more than 3 service labels, the command will not do anything.
+	`,
+	Args: cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		prNumber := args[0]
+		fmt.Println("PR Number: ", prNumber)
+
+		gh := github.NewClient()
+		execRequestServiceReviewers(prNumber, gh, labeler.EnrolledTeamsYaml)
+	},
+}
+
+func execRequestServiceReviewers(prNumber string, gh GithubClient, enrolledTeamsYaml []byte) {
+	pullRequest, err := gh.GetPullRequest(prNumber)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	enrolledTeams := make(map[string]labeler.LabelData)
+	if err := yaml.Unmarshal(enrolledTeamsYaml, &enrolledTeams); err != nil {
+		fmt.Printf("Error unmarshalling enrolled teams yaml: %s", err)
+		os.Exit(1)
+	}
+
+	requestedReviewers, err := gh.GetPullRequestRequestedReviewers(prNumber)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	previousReviewers, err := gh.GetPullRequestPreviousReviewers(prNumber)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	// If more than three service labels are impacted, don't request reviews.
+	// Only request reviews from unique service teams.
+	githubTeams := make(map[string]struct{})
+	teamCount := 0
+	for _, label := range pullRequest.Labels {
+		if !strings.HasPrefix(label.Name, "service/") || label.Name == "service/terraform" {
+			continue
+		}
+		teamCount += 1
+		if labelData, ok := enrolledTeams[label.Name]; ok {
+			githubTeams[labelData.Team] = struct{}{}
+		}
+	}
+
+	if teamCount > 3 {
+		fmt.Println("Provider-wide change (>3 services impacted); not requesting service team reviews")
```

Comment:
An aside..

How are we thinking about service team reviews. 

My time to review tool isn't taking to account this structure.

What if a service team isn't responding to review requests?

Reply:
go/terraform-resource-owner-pr-notifications has details